flow/layers/aiks_layer.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/aiks_layer.h"
+
+#include <utility>
+
+namespace flutter {
+
+AiksLayer::AiksLayer(const SkPoint& offset,
+                     const std::shared_ptr<const impeller::Picture>& picture)
+    : offset_(offset), picture_(picture) {
+#if IMPELLER_SUPPORTS_RENDERING
+  if (picture_) {
+    auto bounds = picture_->pass->GetElementsCoverage(std::nullopt)
+                      .value_or(impeller::Rect());
+    bounds_ = SkRect::MakeXYWH(bounds.origin.x + offset_.x(),
+                               bounds.origin.y + offset.y(), bounds.size.width,
+                               bounds.size.height);
+  }
+#endif
+}
```

Comment:
There is no override for IsReplacing. I forget the details, but sometimes a new <Picture>Layer is constructed with the same ui.Picture as was used in the previous frame and so we need to check if the pictures are the same, not just the layer, otherwise we lose a lot of partial repaint reductionl

Reply:
Indeed. The way it is now partial repaint will not work correctly with `AiksLayer`. With `DisplayListLayer` (and similarly `AiksLayer`) the diffing is moved one level up - to `ContainerLayer`.  The `DisplayListLayer::Diff` basically just reports the layer paint region.

The reason for this is that we want to be able to detect added / removed display list layers similar to how the framework diffs element children. (See `ContainerLayer::DiffChildren`).

Reply:
The reason why picture layers are different than other layers in this regard is that picture layers don't exit in framework in a similar way other layers exist - so we don't get the old layer to link with the new layer when adding picture like we would get with a backdrop filter or a clip rect layer for example. So we relay on comparing the contents.